packages/main/src/Menu.ts

```diff
@@ -362,335 +248,114 @@ class Menu extends UI5Element {
+		}, MENU_OPEN_DELAY);
 	}
 
-	_itemMouseOut(e: MouseEvent) {
-		if (isDesktop()) {
-			const opener = e.target as OpenerStandardListItem;
-			const item = opener.associatedItem;
+	_itemClick(e: CustomEvent<ListItemClickEventDetail>) {
+		const item = e.detail.item as MenuItem;
 
-			clearTimeout(this._timeout);
+		if (!item._popover) {
+			const prevented = !this.fireEvent<MenuItemClickEventDetail>("item-click", {
+				"item": item,
+				"text": item.text,
+			}, true, false);
 
-			// Close submenu with 400ms delay
-			if (item && item.hasSubmenu && item._subMenu) {
-				// try to close the sub-menu
-				item._preventSubMenuClose = false;
-				this._startCloseTimeout(item);
+			if (!prevented && this._popover) {
+				item.fireEvent("close-menu", {});
 			}
+		} else {
+			this._openItemSubMenu(item);
 		}
 	}
 
 	_itemKeyDown(e: KeyboardEvent) {
 		const shouldCloseMenu = this.isRtl ? isRight(e) : isLeft(e);
 		const shouldOpenMenu = this.isRtl ? isLeft(e) : isRight(e);
+		const item = e.target as MenuItem;
+		const parentElement = item.parentElement as MenuItem;
 
 		if (isEnter(e)) {
 			e.preventDefault();
 		}
 		if (shouldOpenMenu) {
-			const opener = e.target as OpenerStandardListItem;
-			const item = opener.associatedItem;
-
-			item.hasSubmenu && this._prepareSubMenu(item, opener);
-		} else if (shouldCloseMenu && this._isSubMenu && this._parentMenuItem) {
-			const parentMenuItemParent = this._parentMenuItem.parentElement as Menu;
-			parentMenuItemParent._closeItemSubMenu(this._parentMenuItem, true, true);
-		}
-	}
-
-	_itemClick(e: CustomEvent<ListItemClickEventDetail>) {
-		const opener = e.detail.item as OpenerStandardListItem;
-		const item = opener.associatedItem;
-
-		if (!item.hasSubmenu) {
-			// click on an item that doesn't have sub-items fires an "item-click" event
-			if (!this._isSubMenu) {
-				// fire event if the click is on top-level menu item
-				const prevented = !this.fireEvent<MenuItemClickEventDetail>("item-click", {
-					"item": item,
-					"text": item.text,
-				}, true, false);
-
-				if (!prevented) {
-					this._popover!.open = false;
-				}
-			} else {
-				const mainMenu = this._findMainMenu(item);
-				const prevented = !mainMenu.fireEvent<MenuItemClickEventDetail>("item-click", {
-					"item": item,
-					"text": item.text,
-				}, true, false);
-
-				if (!prevented) {
-					let openerMenuItem = item;
-					let parentMenu = openerMenuItem.parentElement as Menu;
-					do {
-						openerMenuItem._preventSubMenuClose = false;
-						this._closeItemSubMenu(openerMenuItem);
-						parentMenu = openerMenuItem.parentElement as Menu;
-						openerMenuItem = parentMenu._parentMenuItem as MenuItem;
-					} while (parentMenu._parentMenuItem);
-
-					mainMenu._popover!.preventFocusRestore = false;
-					mainMenu._popover!.open = false;
-				}
+			this._openItemSubMenu(item);
+		} else if (shouldCloseMenu && parentElement.hasAttribute("ui5-menu-item")) {
+			if (parentElement._popover) {
```

Comment:
We can move this condition to the 'else if' condition above and prevent nesting 'if' statements.

Reply:
done